src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,362 @@
+365.25. The default year starts in the 10th dekad (1st of
+April) until the last dekad of March, to correspond with the
+West African monsoon season.
+"""
+
+import os
+from pathlib import Path
+from typing import List, Literal, Union, get_args
+
+import numpy as np
+import pandas as pd
+import rioxarray
+
+from src.utils_general.utils import download_ftp
+
+_BASE_URL = (
+    "http://80.69.76.253:8080/geoserver"
+    "/Biomass/wfs?&REQUEST="
+    "GetFeature&SERVICE=wfs&VERSION=1.1.0"
+    "&TYPENAME=WA_DMP_{admin}_ef_v0&"
+    "outputformat=CSV&srsName=EPSG:4326"
+)
+
+_RAW_FILENAME = "WA_DMP_{admin}_ef_v0.csv"
+
+_RAW_PATH = Path(os.getenv("AA_DATA_DIR"), "public", "raw", "glb", "biomasse")
+
+
+def _get_raw_path(admin):
+    return Path(_RAW_PATH, _RAW_FILENAME.format(admin=admin))
+
+
+_PROCESSED_FILENAME = "biomasse_{iso3}_{admin}_dekad_{start_dekad}.csv"
+
+
+def _get_processed_path(admin, start_dekad, iso3: Union[None, str] = None):
+    if iso3 is None:
+        iso3 = "glb"
+
+    _processed_path = Path(
+        os.getenv("AA_DATA_DIR"), "public", "processed", iso3, "biomasse"
+    )
+
+    return Path(
+        _processed_path,
+        _PROCESSED_FILENAME.format(
+            iso3=iso3, admin=admin, start_dekad=start_dekad
+        ),
+    )
+
+
+AdminArgument = Literal["ADM0", "ADM1", "ADM2"]
+_VALID_ADMIN = get_args(AdminArgument)
+
+
+def _check_admin(admin):
+    if admin not in _VALID_ADMIN:
+        raise ValueError("`admin` must be one of 'ADM0', 'ADM1', or 'ADM2'.")
+
+
+def download_dmp(admin: AdminArgument = "ADM2"):
+    """Download raw DMP data
+
+    Raw DMP data is downloaded for specified
+    administrative area. This data is downloaded
+    for all countries in West Africa covered by
+    Biomasse, using those specific countries'
+    administrative boundaries, and is downloaded
+    for all available years. The data is downloaded
+    in CSV format and can be processed using
+    ``process_dmp()``.
+    """
+    _check_admin(admin)
+    url = _BASE_URL.format(admin=admin)
+    raw_path = _get_raw_path(admin)
+    download_ftp(url=url, save_path=raw_path)
+    return None
+
+
+def load_dmp(admin: AdminArgument = "ADM2", redownload: bool = False):
+    _check_admin(admin)
+    raw_path = _get_raw_path(admin)
+    raw_path.unlink(missing_ok=True)
+    download_dmp(admin)
+
+    # na values set by Biomasse as -9998.8
+    df = pd.read_csv(raw_path, na_values=["-9998.8"])
+    df.dropna(axis="columns", how="all", inplace=True)
+    return df
+
+
+def process_dmp(
+    admin: AdminArgument = "ADM2",
+    redownload: bool = False,
+    start_dekad: int = 10,
+):
+    """Process DMP raw data
+
+    DMP raw data is received in a wide format
+    dataset that needs processing. This pivots
+    the data from wide to long format with
+    DMP dekadal observations for all years, while
+    also pivoting the mean values to allow for
+    calculation of biomasse anomaly.
+    """
+    df = load_dmp(admin, redownload)
```

Comment:
I would assume the dataset is already downloaded when calling process. So check for file existence and if doesn't exist raise an error.